src/checks/y_check_prefer_is_not.clas.abap

```diff
@@ -74,5 +82,8 @@ CLASS y_check_prefer_is_not IMPLEMENTATION.
                       OR token-str CP 'MATCHES*' ).
   ENDMETHOD.
 
+  METHOD is_predicative_method.
+    result = xsdbool( token-str CP '*(' ).
+  ENDMETHOD.
```

Comment:
Unfortunately, this is too simple and also recognizes non-predicative calls, as in:
```abap
report test.

class main definition.
  public section.
    class-methods get_number
      returning value(number) type i.
endclass.

CLASS main IMPLEMENTATION.

  METHOD get_number.
  ENDMETHOD.

ENDCLASS.

start-of-selection.
  if not main=>get_number( ) = 1. "<---there should be a finding here
  endif.
```

Reply:
Oh no, I forgot about that... let me see if I can fix that.

Reply:
Can you think of other cases that aren't covered by the reworked logic?

Reply:
Well, now I understand why nobody attempted to solve this issue in the past two and a half years. 😅 Thank you for your help!